### PR TITLE
Dependencies

### PR DESCRIPTION
**Pull request by etrobert**

### CODE REVIEW

File: src/App/Dependency/Dependency.tsx [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+import { DependencyId, dependencyPathSelectorFamily } from "../atoms";
+
+type Props = {
+  id: DependencyId;
+};
+
+const Dependency = ({ id }: Props): JSX.Element => {
+  const path = useRecoilValue(dependencyPathSelectorFamily(id));
+
+  return <path d={path} />;
```
Comment (justinline) [inline src/App/Dependency/Dependency.tsx:12]: 😍

File: src/App/Task/Task.tsx [hunk 2]
```diff
@@ -13,8 +17,16 @@ const Task = ({ id }: Props): JSX.Element => {
     name,
     status,
   } = useRecoilValue(taskStateFamily(id));
+
+  const setBoxSize = useSetRecoilState(taskBoxSizeStateFamily(id));
+
+  const ref = useRef<HTMLDivElement>(null);
+  const boxSize = useBoxSizeObserver(ref);
+  useEffect(() => boxSize && setBoxSize(boxSize), [boxSize, setBoxSize]);
```
Comment (justinline) [inline src/App/Task/Task.tsx:25]: thought: i've seen you use `&&` so many times in these reviews a couple weeks after you ranted about it 😂

File: src/useBoxSizeObserver.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
+import { RefObject, useEffect, useMemo, useState } from "react";
+import { BoxSize } from "./App/atoms";
+
+const useBoxSizeObserver = <T extends HTMLElement>(
+  ref: RefObject<T>
+): BoxSize | undefined => {
+  const [boxSize, setBoxSize] = useState<BoxSize>();
+
+  const ro = useMemo(
```
Comment (justinline) [inline src/useBoxSizeObserver.ts:9]: suggestion: rename to `memoizedObserver` or just `observer` or something, had to look into it to understand what it was

File: src/useBoxSizeObserver.ts [hunk 4]
```diff
@@ -0,0 +1,35 @@
+import { RefObject, useEffect, useMemo, useState } from "react";
+import { BoxSize } from "./App/atoms";
+
+const useBoxSizeObserver = <T extends HTMLElement>(
```
Comment (justinline) [inline src/useBoxSizeObserver.ts:4]: praise: nice hook :)

**Comment by justinline** [comment]
Finshed 🏁 Thanks for following it up :D

File: src/App/Task/Task.tsx (hunk 2 above)
Comment (etrobert) [reply src/App/Task/Task.tsx:25]: Hahaha

Well from the doc it only matters if you return a function... But you're right it's confusing if you don't know the API of `useEffect` in details (I had to look it up to be sure)

I'll change it to make it more explicit 😁

File: src/useBoxSizeObserver.ts (hunk 3 above)
Comment (etrobert) [reply src/useBoxSizeObserver.ts:9]: 🙈 

Will do. This is part of the 2am code.

**Pull request merged**